packages/stripe/src/Adaptor.js

```diff
@@ -0,0 +1,313 @@
+import { expandReferences } from '@openfn/language-common/util';
+import * as util from './Utils';
+
+/**
+ * State object
+ * @typedef {Object} HttpState
+ * @property data - the parsed response body
+ * @property response - the response from the HTTP server, including headers, statusCode, body, etc
+ * @property references - an array of all previous data objects used in the Job
+ **/
+
+/**
+ * Payment Intents Options object. See {@link https://docs.stripe.com/api/payment_intents/list|Stripe Payment Intents API documentation} for more details.
+ * @typedef {Object} PaymentIntentsOptions
+ * @public
+ * @property {object} created - An object of created fields for filtering the data.
+ * @property {string} ending_before - Object string ID that defines your place in the list. Example: `obj_bar` in order to fetch the previous page of the list.
+ * @property {string} starting_after - Object string ID that defines your place in the list. Example: `obj_foo` in order to fetch the next page of the list.
+ * @property {number} limit - A limit on the number of objects to be returned. Limit can range between 1 and 100, and the default is 10.
+ */
+
+/**
+ * Get payments from Stripe
+ * @example <caption>List all payment intents</caption>
+ * listPaymentIntents();
+ * @example <caption>List payment intents with options</caption>
+ * listPaymentIntents({limit:3});
+ * @function
+ * @public
+ * @param {PaymentIntentsOptions} options - Optional payment intent request options
+ * @returns {Operation}
+ * @state {HttpState}
+ */
+export function listPaymentIntents(options = {}) {
+  return async state => {
+    const [resolvedoptions] = expandReferences(state, options);
+
+    const response = await util.request(
+      state.configuration,
+      'GET',
+      '/payment_intents',
+      {
+        query: {
+          ...resolvedoptions,
+        },
+      }
+    );
+
+    return util.prepareNextState(state, response);
+  };
+}
+
+/**
+ * Get a Stripe payment intent
+ * @example
+ * getPaymentIntent('pi_3RxS5EEAUr6ipfDB0aKo3moC');
+ * @function
+ * @public
+ * @param {string} paymentIntentId - The ID of the payment intent to retrieve.
+ * @returns {Operation}
+ * @state {HttpState}
+ */
+export function getPaymentIntent(paymentIntentId) {
```

Comment:
Sorry I'm just going to ask again: you're absolutely _sure_ we don't just want `get('payment_intents', $.intentId)`?

I see nothing in these functions that wouldn't work in a generic function, and that gives us better access to the whole Stripe API without much loss in usability.

Reply:
@josephjclark  we can, so we will only have a single `get` and we leave the other `listCustomers()`?

Reply:
@josephjclark  I added `get()` and `list()`. If they are okay I can remove the other functions